src/query.ts

```diff
@@ -345,3 +346,23 @@ export function stringifyEncodingQueryFieldDef(encQ: EncodingQuery): string {
     params.map((p) => ',' + p.key + '=' + p.value).join('');
   return (fn ? fn + '(' + fieldType + ')' : fieldType);
 }
+
+export function scaleType(encQ: EncodingQuery): ScaleType {
```

Comment:
Don't take the whole `encQ` as input. Take `scaleType`, `timeUnit`, and `type` as inputs so when you call it from constraint, it's clear that you only use only these properties of `encQ`.